include/deal.II/matrix_free/evaluation_template_factory.h

```diff
@@ -60,6 +60,9 @@ namespace internal
       VectorizedArrayType *gradients_quad,
       VectorizedArrayType *scratch_data,
       const bool           sum_into_values_array);
+
+    static unsigned int
+    max_precompiled_degree();
```

Comment:
You could even make this function `constexpr`, and then it could be used to allocate arrays, for example.